bitcoin/src/address.rs

```diff
@@ -616,7 +616,7 @@ mod sealed {
 
 /// Marker of status of address's network validation. See section [*Parsing addresses*](Address#parsing-addresses)
 /// on [`Address`] for details.
-pub trait NetworkValidation: sealed::NetworkValidation {
+pub trait NetworkValidation: sealed::NetworkValidation + Sync + Send + Sized {
```

Comment:
ACK adding `Send + Sync`, also `Unpin` would be nice. `Sized` shouldn't be needed but I guess I don't care.

Regarding serde wouldn't it be simpler to do this in `sealed` module?

```rust
#[cfg(feature = "serde")]
pub trait NetworkValidation:  serde::Serialize {}


#[cfg(not(feature = "serde"))]
pub trait NetworkValidation:  serde::Serialize {}
```

Reply:
Why? I don't see why the serde implementation should be dependent on the network validation at all. Any address is serializable whether or not you know the network it's for or not, right?

Reply:
Oh, crap, I meant to do the same as the other traits but I it'd have to be `where Address<Self>: Serialize` but maybe we would get a cycle in bounds. If it doesn't work then I guess your changes make sense.